pkg/commands/kustomizationfile.go

```diff
@@ -30,9 +34,18 @@ import (
 	"github.com/kubernetes-sigs/kustomize/pkg/types"
 )
 
+var kustomizationFields = []string{"resources", "bases", "namePrefix", "namespace", "crds", "commonLabels", "commonAnnotations", "patches", "configMapGenerator", "secretGenerator", "vars", "imageTags"}
+
+type fieldComment struct {
+	field   string
+	comment []byte
+}
+
 type kustomizationFile struct {
-	path string
-	fsys fs.FileSystem
+	path     string
+	fsys     fs.FileSystem
+	comments []fieldComment
+	fields   map[string]bool
```

Comment:
isn't this redundant?  i.e.
```
 func (mf *kustomizationFile) hasField(name string) bool {
  for _, n := range mf.originalFields {
    if n == name {
     return true
    }
  }
  return false
}
```

Reply:
Yes, we can do this by a for loop.  I use a map to avoid this loop :)

Reply:
two data structures instead of one introduces the possibility that they mean different things, and can get out of sync as code changes.  performance not an issue here.

Reply:
done